### ISSUE TITLE
Confusing behavior for datashade and redim

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#1992]
If I call redim.range on a plot *before* specifying the datashader operator, it works as expected:

```
import pandas as pd
import numpy as np
import datashader as ds
import datashader.transfer_functions as tf
import holoviews as hv
import holoviews.operation.datashader as hvd

hvd.shade.cmap=["lightblue", "darkblue"]
hv.extension("bokeh") 
np.random.seed(1)

df = pd.DataFrame.from_items([('x',np.random.normal(0,2,10000)),
                              ('y',np.random.normal(0,2,10000))])

cvs = ds.Canvas(plot_width=300, plot_height=300, x_range=(-5,5), y_range=(-5,5))
agg = cvs.points(df, 'x', 'y')
tf.shade(agg)
```
![image](https://user-images.githubusercontent.com/1695496/31397700-55936258-adac-11e7-8467-f7c4ab067118.png)

```
plot = hvd.datashade(hv.Points(df).redim.range(x=(-5,5),y=(-5,5)))
plot.opts(plot=dict(height=300,width=300))
```
![image](https://user-images.githubusercontent.com/1695496/31397719-612b4dec-adac-11e7-88da-c6300ff9e9b8.png)

But if I call it afterwards, the first plot shown is at a bogus low resolution, presumably because it has been snipped out of a datashader rendering of the entire available plot range:

```
plot = hvd.datashade(hv.Points(df))
plot.opts(plot=dict(height=300,width=300)).redim.range(x=(-5,5),y=(-5,5))
```

![image](https://user-images.githubusercontent.com/1695496/31397769-7e6e3752-adac-11e7-8ef2-a5c05570bbe6.png)

The plot still updates correctly when zooming in at this point, but it would be nice if the plot could automatically update after rendering to respect the provided range, just as it already does to respect the plot size.

**Comment by timothymillar** [comment]
@jbednar I believe this is another instance of the issue I have run into,
I have been trying to programmatically write out multiple plots that contain datashader layers and trying to avoid rendering them in a notebook first (there is quite a few of them).

My data are best viewed with a long x-axis relative to y-axis so I scale the components with `.opts(plot=dict(width=1200, height=300))` but the  datashader layer(s) seem to be initially rendered at the default dimensions (a square) and not redrawn before they are scaled so the resulting image is low resolution and stretched:
![screen shot 2017-12-20 at 1 54 22 pm](https://user-images.githubusercontent.com/14065102/34185960-799b3284-e58d-11e7-8952-0e41ebe182c3.png)


If I render the plot in the notebook *before* I save it to disk then the datashader layer is updated and renders correctly in the output, but this is not the ideal method:
![screen shot 2017-12-20 at 1 54 44 pm](https://user-images.githubusercontent.com/14065102/34185971-8cfbcc4e-e58d-11e7-8902-b86ecf446b5b.png)


Example code:
```
points = hv.Points(dataframe, 
                   kdims=['position', 'score'])

plot = datashade(points, 
                 y_range=(-1, 1), 
                 color_key=pallet,
                 aggregator=ds.count_cat('group')).opts(plot=dict(width=1200, height=300))

renderer = hv.renderer('bokeh')
renderer.save(plot, 'example')
```

Is there another way to force the datashader layer to redraw without rendering it in the notebook?

**Comment by philippjfr** [comment]
@timothymillar No, the PlotSize stream waits on bokeh to display the plot to figure out the appropriate width and height for it. I'd suggest setting a higher default width and height:

```python
plot = datashade(points, 
                 y_range=(-1, 1),
                 width=1200, height=300, 
                 color_key=pallet,
                 aggregator=ds.count_cat('group')).opts(plot=dict(width=1200, height=300))

**Comment by timothymillar** [comment]
@philippjfr thanks for the quick response (and for holoviews)!
That works perfectly, don't know how I missed it...
